streams/src/main/java/org/apache/kafka/streams/kstream/internals/graph/UnoptimizableRepartitionNode.java

```diff
@@ -0,0 +1,184 @@
+import org.apache.kafka.streams.processor.internals.InternalTopologyBuilder;
+
+/**
+ * Repartition node that is not subject of optimization algorithm
+ */
+public class UnoptimizableRepartitionNode<K, V> extends BaseRepartitionNode<K, V> {
+    private final StreamPartitioner<K, V> partitioner;
+    private final InternalTopicProperties internalTopicProperties;
+
+    private UnoptimizableRepartitionNode(final String nodeName,
+                                         final String sourceName,
+                                         final ProcessorParameters processorParameters,
+                                         final Serde<K> keySerde,
+                                         final Serde<V> valueSerde,
+                                         final String sinkName,
+                                         final String repartitionTopic,
+                                         final StreamPartitioner<K, V> partitioner,
+                                         final InternalTopicProperties internalTopicProperties) {
+        super(
+            nodeName,
+            sourceName,
+            processorParameters,
+            keySerde,
+            valueSerde,
+            sinkName,
+            repartitionTopic
+        );
+
+        this.partitioner = partitioner;
+        this.internalTopicProperties = internalTopicProperties;
+    }
+
+    @Override
+    Serializer<V> getValueSerializer() {
+        return valueSerde != null ? valueSerde.serializer() : null;
+    }
+
+    @Override
+    Deserializer<V> getValueDeserializer() {
+        return valueSerde != null ? valueSerde.deserializer() : null;
+    }
+
+    @Override
+    public void writeToTopology(final InternalTopologyBuilder topologyBuilder) {
+        final Serializer<K> keySerializer = getKeySerializer();
+        final Deserializer<K> keyDeserializer = getKeyDeserializer();
+
+        topologyBuilder.addInternalTopic(repartitionTopic, internalTopicProperties);
+
+        topologyBuilder.addProcessor(
+            processorParameters.processorName(),
+            processorParameters.processorSupplier(),
+            parentNodeNames()
+        );
+
+        topologyBuilder.addSink(
+            sinkName,
+            repartitionTopic,
+            keySerializer,
+            getValueSerializer(),
+            partitioner,
+            processorParameters.processorName()
+        );
+
+        topologyBuilder.addSource(
+            null,
+            sourceName,
+            new FailOnInvalidTimestamp(),
+            keyDeserializer,
+            getValueDeserializer(),
+            repartitionTopic
+        );
+    }
+
+    private Serializer<K> getKeySerializer() {
+        return keySerde != null ? keySerde.serializer() : null;
+    }
+
+    private Deserializer<K> getKeyDeserializer() {
+        return keySerde != null ? keySerde.deserializer() : null;
+    }
+
+    public static <K, V> UnoptimizableRepartitionNodeBuilder<K, V> repartitionNodeBuilder() {
```

Comment:
I can change it, sure but personally i would prefer to have the builder here because 1) it follows same standard as other `BaseRepartitionNode` implementations 2) static factory method for the `UnoptimizableRepartitionNodeBuilder` will have a lot of parameters and it'll make code uglier.
wdyt?

Reply:
Fair enough. Thanks for the reply.